graylog2-web-interface/src/views/components/aggregationbuilder/SeriesConfiguration.jsx

```diff
@@ -21,17 +21,22 @@ export default class SeriesConfiguration extends React.Component {
 
   _onSubmit = () => {
     const { series, name } = this.state;
-    if (name && name !== series.effectiveName) {
+    const { onClose } = this.props;
+    const newName = name || series.function;
+
+    if (newName && newName !== series.effectiveName) {
       const { config } = series;
```

Comment:
Why only destruct `config` and not `function` and `effectiveName`?

Reply:
Good point, I've implemented:
`const { name, series, series: { config, function: functionName, effectiveName } } = this.state;`
to deconstruct the series object, while still being able to run `series.toBuilder()`